src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -1899,7 +1887,37 @@ private ConsumeTokenResult ConsumeNextTokenFromLastNonCommentTokenMultiSegment()
             }
             else
             {
-                goto RollBack;
+                Debug.Assert(_tokenType == JsonTokenType.EndArray || _tokenType == JsonTokenType.EndObject);
+                if (_inObject)
+                {
+                    Debug.Assert(first != JsonConstants.CloseBracket);
```

Comment:
CloseBracket, or CloseBrace?  (Both array and object are asserting the same thing, seems one of them should be different)

Reply:
Yes, the `Debug.Assert` should say `CloseBrace` in this object case. Good catch.

Reply:
Technically, it could say both/either since both `first == JsonConstants.CloseBrace` and `first == JsonConstants.CloseBracket` conditions are checked previously. For clarity, will update just for the single constant.

Reply:
Added to master as well: https://github.com/dotnet/corefx/pull/40454